x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -1155,6 +1164,78 @@ private void assertSearchableSnapshotStats(String indexName, boolean cacheEnable
         }
     }
 
+    public void testMountingSnapshotLinksRefreshTaskAsChild() throws Exception {
+        final CyclicBarrier cyclicBarrier = new CyclicBarrier(2);
+
+        final String indexName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        createAndPopulateIndex(indexName, indexSettingsNoReplicas(1).put(INDEX_SOFT_DELETES_SETTING.getKey(), true));
+
+        final String repositoryName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        createRepository(repositoryName, "fs");
+
+        final SnapshotId snapshotOne = createSnapshot(repositoryName, "snapshot-1", List.of(indexName)).snapshotId();
+        assertAcked(indicesAdmin().prepareDelete(indexName));
+
+        // block the cluster state update thread, so we have the opportunity to inspect the mount/restore tasks' metadata
+        ClusterService clusterService = internalCluster().getCurrentMasterNodeInstance(ClusterService.class);
+        clusterService.submitUnbatchedStateUpdateTask("block master service", new ClusterStateUpdateTask() {
+            @Override
+            public ClusterState execute(ClusterState currentState) {
+                safeAwait(cyclicBarrier);   // notify test that we're blocked
+                safeAwait(cyclicBarrier);   // wait to be unblocked by test
+                return currentState;
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                throw new AssertionError("block master service", e);
+            }
+        });
+
+        // wait for master thread to be blocked
+        safeAwait(cyclicBarrier);
+
+        ActionFuture<RestoreSnapshotResponse> response = null;
+        try {
+            final MountSearchableSnapshotRequest mountRequest = new MountSearchableSnapshotRequest(
+                indexName,
+                repositoryName,
+                snapshotOne.getName(),
+                indexName,
+                Settings.EMPTY,
+                Strings.EMPTY_ARRAY,
+                true,
+                MountSearchableSnapshotRequest.Storage.FULL_COPY
+            );
+
+            response = client().execute(MountSearchableSnapshotAction.INSTANCE, mountRequest);
+            TaskInfo[] restoreSnapshotTask = new TaskInfo[1];
+            waitUntil(() -> {
+                restoreSnapshotTask[0] = getTaskForActionFromMaster(TransportRestoreSnapshotAction.TYPE.name());
+                return restoreSnapshotTask[0] != null;
+            });
+            TaskInfo mountSnapshotTask = getTaskForActionFromMaster(MountSearchableSnapshotAction.NAME);
+            assertNotNull("No mount task found", mountSnapshotTask);
+            assertEquals(mountSnapshotTask.taskId(), restoreSnapshotTask[0].parentTaskId());
+        } finally {
+            safeAwait(cyclicBarrier);   // Unblock the master thread
+            if (response != null) {
+                response.actionGet();
+                assertAcked(indicesAdmin().prepareDelete(indexName));
```

Comment:
The `actionGet` and delete index here were an attempt to fix [a failure that occurred in CI in a teardown](https://gradle-enterprise.elastic.co/s/762xzz33sxoqg/tests/task/:x-pack:plugin:searchable-snapshots:internalClusterTest/details/org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests/testMountingSnapshotLinksRefreshTaskAsChild?expanded-stacktrace=WyIwLTEiXQ&top-execution=1), one or both of these may be necessary. Happy to remove the delete if it's not necessary.

Reply:
Nice catch, yeah, it seems we have to wait for the mount to complete before finishing the test or else there's a race between cleaning up the test cluster and the mount process creating the internal `.snapshot-blob-cache` index.

I expect there's no need to manually delete the test index here tho but it doesn't hurt either. Would you add a code comment about why we need these nonobvious cleanup steps?